src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2450,7 +2450,21 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatement(SyntaxKind par
                     }
 
                     var incompleteMember = _syntaxFactory.IncompleteMember(attributes, modifiers.ToList(), type.IsMissing ? null : type);
-                    if (incompleteMember.ContainsDiagnostics)
+
+                    var containsErrorDiagnostic = false;
+                    foreach (var child in incompleteMember.EnumerateNodes())
```

Comment:
What happens though if the diagnostic is in a grandchild of the original node?  We'd miss that here. 

Still suspicious that this is the correct approach.  I dug through the parser and couldn't find any other places where we call `EnumerateNodes` or check `DiagnosticSeverity`.  

Perhaps @gafter has a recommendation on this approach.